### ISSUE TITLE
Can't make certain sample loop (regression from #35912)

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#36289]
**Godot version:**
7724b842f

**Steps to reproduce:**
1. Download this sample: https://cdn.discordapp.com/attachments/477544613511692358/678742615662919702/Waterfall.ogg
2. Import it in your project and make sure looping is enabled
3. Add it to some AudioStreamPlayer and play it
4. But where's the loop ;_;

This works fine in 3.2, doesn't seem to happen with other samples.

**Comment by Anutrix** [comment]
The sample seems to pause at 7.29 sec and not reach 7.31 sec. Playing it again stops it at 7.31.

Another thing I found out was that when I click anywhere on the spectrogram(thus skipping a bit of it) on the bottom part of audio sampler while it is playing, it starts to loop properly.
The problem occurs again if I stop and start it again.

**Comment by avsoftlabs** [comment]
This seems to be in 3.2 stable - after migrating from one of the 3.2 betas audio has sometimes not looping when it should be. I never observed this behaviour in any version prior to 3.2 stable.

After experiencing it in a project (Godot 3.2.stable.official, Win64) I tried to make a minimal reproduction project to post here only  to find that the exact same sample worked in that project, and that I can't seem to reliably reproduce the bug. I'll keep trying to reproduce the issue and and post it here.

**Comment by avsoftlabs** [comment]
Whoops, I was accidentally testing in 3.2.1_rc, NOT 3.2 stable. This is reproducible in 3.2.1_rc, at least on Win64. Example:
[https://github.com/avsoftlabs/godotaudiolooptest](https://github.com/avsoftlabs/godotaudiolooptest)

3.2 stable doesn't appear to have this bug

**Comment by Calinou** [comment]
@avsoftlabs You could try to bisect the regression. I wrote a guide for it here, but it's not merged yet: https://github.com/godotengine/godot-docs/pull/3112

**Comment by akien-mga** [comment]
Possibly a regression from #35912, which was cherry-picked in the 3.2 branch (I didn't test, but looking over commits it's the most likely culprit).

**Comment by bruce965** [comment]
My apologies for this regression. The following change fixes it without reopening #35912.

Should I open a pull request?

```diff
diff --git a/modules/stb_vorbis/audio_stream_ogg_vorbis.cpp b/modules/stb_vorbis/audio_stream_ogg_vorbis.cpp
index f2d0f5c9a6..606de8921b 100644
--- a/modules/stb_vorbis/audio_stream_ogg_vorbis.cpp
+++ b/modules/stb_vorbis/audio_stream_ogg_vorbis.cpp
@@ -57,7 +57,8 @@ void AudioStreamPlaybackOGGVorbis::_mix_internal(AudioFrame *p_buffer, int p_fra
 
                if (todo) {
                        //end of file!
-                       if (vorbis_stream->loop && mixed > 0) {
+                       bool is_not_empty = mixed > 0 || stb_vorbis_stream_length_in_samples(ogg_stream) > 0;
+                       if (vorbis_stream->loop && is_not_empty) {
                                //loop
                                seek(vorbis_stream->loop_offset);
                                loops++;
```

**Comment by KoBeWi** [comment]
> Should I open a pull request towards master?

Yes.
